pctiler/Dockerfile

```diff
@@ -9,28 +9,22 @@ FROM python:3.8.7-buster
 # exposure.
 ARG DEVOPS_PAT
 
-EXPOSE 8000
-
 RUN apt-get update && \
     apt-get install -y build-essential libgeos++-dev libproj-dev
 
 ENV CURL_CA_BUNDLE /etc/ssl/certs/ca-certificates.crt
 
 RUN pip install --upgrade pip setuptools wheel
 
-RUN mkdir -p /opt/src/pctiler
-
-COPY .isort.cfg /opt/src/.isort.cfg
-COPY pctiler/requirements.txt /opt/src/
-COPY pctiler/requirements.txt /opt/src/requirements.txt
-
-RUN pip install -r /opt/src/requirements.txt
-
-ENV PYTHONPATH=/opt/src/pctiler:$PYTHONPATH
-
 WORKDIR /opt/src
 
-COPY pctiler/templates /opt/src/templates
+COPY .isort.cfg .isort.cfg
+COPY pctiler/gunicorn_conf.py gunicorn_conf.py
+
+COPY pctiler pctiler
+COPY pccommon pccommon
+RUN pip install ./pccommon ./pctiler["server"]
```

Comment:
To explain the previous Docker setup - the way you have it now will always reinstall dependencies whenever source code changes and you run scripts/update. This is inconvenient as it can increase development cycles when needing to rebuild containers. The previous way only installed requirements if the requirements file changed. I'm OK with removing the quicker docker build times for simplification if it makes things easier for this refactor.

I do see that it's not installing an editable copy. The previous version would ensure the PYTHONPATH was set correctly so that mounted volumes means during development source code changes would be reflected in the containers. Is this still true for the current setup? If not, we'll want to make sure that capability survives this refactor.

Reply:
> the way you have it now will always reinstall dependencies whenever source code changes and you run scripts/update. This is inconvenient as it can increase development cycles when needing to rebuild containers. The previous way only installed requirements if the requirements file changed. I'm OK with removing the quicker docker build times for simplification if it makes things easier for this refactor.

Well we could switch back if you want, I just tried to write a more `pythonic` package using setup.py and narrowed requirements.


> I do see that it's not installing an editable copy. The previous version would ensure the PYTHONPATH was set correctly so that mounted volumes means during development source code changes would be reflected in the containers. Is this still true for the current setup? If not, we'll want to make sure that capability survives this refactor.

Yeah, this is kinda dangerous to just copy and not install the package. It could get messy if you have generated the `pyc` files locally and for some reason the python within the docker would use them (by experience it never happens but I prefer avoiding risk). 

> The previous version would ensure the PYTHONPATH was set correctly so that mounted volumes

Yeah, it also means that the docker image is really linked to the docker-compose definition (mounted volume) which is also kinda tricky. I really had hard time when starting coding on the project because of the different COPY and mounted volume, IMO the `new` way is simpler, thus maybe better to onboard new devs 🤷‍♂️